### DISCUSSION TITLE
Using getters with parameters needs to be simpler

### DISCUSSION THREAD

**Discussion by aimeos** [vuejs/pinia#2141] [Ideas]
Currently, getters are only computed properties with very limited used in real world applications and if you need to pass parameters, things getting very complicated compared to standard JS objects and functions. Example:

```javascript
  getters: {
    isUsed: (state) => {
      return (key, what) => {
        return state.used[key][what]
      }
    }
  }
```

Using getters with parameters in Vue methods is even more complicated:

```javascript
  import { useSideStore } from '../stores'
  import { storeToRefs } from 'pinia'

    setup() {
      const aside = useSideStore()
      return { aside }
    },
    methods: {
      show(content) {
        const { isUsed } = storeToRefs(this.aside)
        isUsed.value('type', content._type)
      }
    }
```

Instead, using getters with parameters should be as easy as using actions:

```javascript
  // Definition:
  getters: {
    isUsed(key, what) {
      return this.used[key][what]
    }
  }

  // Usage:
  show(content) {
    return useSideStore().isUsed('type', content._type)
  }
```

**Comment by cannap** [comment]
maybe set params via state?

**Reply by aimeos** [reply]
This is possible for sure but then you have an additional state in your store you don't want, is error prone and results in code like:
```javascript
this.aside.params = {key: '123', what: 'sometype'}
return useSideStore().isUsed()
```

**Comment by posva** [comment]
Unfortunately, such getters cannot be cached and are the same as actions.

**Reply by aimeos** [reply]
If actions can be also used as getters, the wording "getters" and "actions" is misleading and should be "computed" and "methods" like in the VueJS core instead.

**Reply by posva** [reply]
Yes, but this is more complex than naming things in an isolated context as it should also feel familiar to Vuex. The naming is a topic that has been revisited countless times, both publicly and internally. We are open to renaming stuff but currently, this would create more confusion than it removes and would definitely need a proper RFC with quite some research work done on it, which of course, feel free to do if this is a topic important to you. I won't hide from you it's quite the amount of work though.

Note that in setup stores, the naming isn't there anymore as you do create computed properties and methods 🙂
